C/c4Document.cc

```diff
@@ -486,12 +486,12 @@ bool c4doc_save(C4Document *doc,
 bool c4doc_setExpiration(C4Database *db, C4Slice docId, uint64_t timestamp, C4Error *outError)
 {
     try {
-        if (!db->get(docId, KeyStore::kMetaOnly).exists()) {
-            recordError(ForestDBDomain, FDB_RESULT_KEY_NOT_FOUND, outError);
+        if (!c4db_beginTransaction(db, outError)) {
             return false;
         }
 
-        if (!c4db_beginTransaction(db, outError)) {
+        if (!db->get(docId, KeyStore::kMetaOnly).exists()) {
```

Comment:
Now the transaction doesn't end if the doc doesn't exist :(

You might want to structure this the way `c4raw_put` does — the `endTransaction` call is after the `catch`, and control always falls through to it. That's safer than trying to make sure to call `endTransaction` before every `return` statement.